### DISCUSSION TITLE
How to use weld to fix a cable on the link?

### DISCUSSION THREAD

**Discussion by Wangzihao12ous** [google-deepmind/mujoco#2005] [Asking for Help]
I used weld to fix a cable on the link2 , but in simulate if i  apply force on the cable ,the cable will be separated from the link2.
Here is the code:
        <weld name="right_boundary0" body1="0B_last" body2="link2_cable_1" anchor="0 0 0" />
        <weld name="right_boundary1" body1="1B_last" body2="link2_cable_2" anchor="0 0 0" />
        <weld name="right_boundary2" body1="2B_last" body2="link2_cable_3" anchor="0 0 0" />

**Comment by Wangzihao12ous** [comment]
I used weld to fix a cable on the link2 , but in simulate if i  apply force on the cable ,the cable will be separated from the link2.
Here is the code:
`    <equality>
        <weld name="right_boundary0" body1="0B_last" body2="link2_cable_1" anchor="0 0 0" />
        <weld name="right_boundary1" body1="1B_last" body2="link2_cable_2" anchor="0 0 0" />
        <weld name="right_boundary2" body1="2B_last" body2="link2_cable_3" anchor="0 0 0" />
    </equality>`
The gray ball is the weld point.
![微信图片_20240327201241](https://github.com/google-deepmind/mujoco/assets/139871867/8eddc54c-1700-401a-9d6a-958aaeb7140a)

**Comment by Wangzihao12ous** [comment]
And how to make the cable through the small hole in link?
The cable have no gemo name,so i cant use contact pair.
![微信图片_20240327202024](https://github.com/google-deepmind/mujoco/assets/139871867/50234a51-75d0-4e0f-8bc2-d16acc66d530)

**Comment by yuvaltassa** [comment]
why are you not using tendons?

**Comment by Wangzihao12ous** [comment]
> why are you not using tendons?

Thank you! I need to use rigid cable in the simulation，ensure that the force on the cable is consistent and there is as little deformation as possible.

**Comment by yuvaltassa** [comment]
And which part of this do you think you cannot do with a tendon?

**Comment by Wangzihao12ous** [comment]
> And which part of this do you think you cannot do with a tendon?

Thank you!In the model I hope to achieve, I can change the position of the first site where the tendon is located, in order to change the length of the tendon. I used muscles and general as actuator, but the results were not very good.
